client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -0,0 +1,195 @@
+			fill="#069E08"
+			fill-opacity="0.35"
+		/>
+	</SVG>
+);
+
+const staging = (
+	<SVG width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+		<Path
+			d="M10.5858 5.63418C11.3668 4.85314 12.6332 4.85314 13.4142 5.63418L18.364 10.5839C19.145 11.365 19.145 12.6313 18.364 13.4124L13.4142 18.3621C12.6332 19.1432 11.3668 19.1432 10.5858 18.3621L5.63604 13.4124C4.85499 12.6313 4.85499 11.365 5.63604 10.5839L10.5858 5.63418Z"
+			fill="#FFB514"
+		/>
+		<Path
+			d="M4.22183 9.16978L9.17157 4.22003C10.7337 2.65793 13.2663 2.65793 14.8284 4.22003L13.4142 5.63424C12.6332 4.85319 11.3668 4.85319 10.5858 5.63424L5.63604 10.584L5.49931 10.7359C4.85838 11.5215 4.90372 12.6801 5.63604 13.4124L10.5858 18.3622L10.7377 18.4989C11.471 19.0972 12.529 19.0972 13.2623 18.4989L13.4142 18.3622L18.364 13.4124C19.0963 12.6801 19.1416 11.5215 18.5007 10.7359L18.364 10.584L19.7782 9.16978C21.3403 10.7319 21.3403 13.2645 19.7782 14.8266L14.8284 19.7764C13.2663 21.3385 10.7337 21.3385 9.17157 19.7764L4.22183 14.8266C2.65973 13.2645 2.65973 10.7319 4.22183 9.16978ZM19.7782 9.16978L18.364 10.584L13.4142 5.63424L14.8284 4.22003L19.7782 9.16978Z"
+			fill="#FFB514"
+			fillOpacity="0.35"
+		/>
+	</SVG>
+);
+
+const Environment = ( { env }: { env: Env } ) => {
+	if ( env === 'staging' ) {
+		return (
+			<HStack justify="flex-start" style={ { width: 'auto', flexShrink: 0 } }>
+				<Icon icon={ staging } />
+				<span>{ __( 'Staging' ) }</span>
+			</HStack>
+		);
+	}
+
+	return (
+		<HStack justify="flex-start" style={ { width: 'auto', flexShrink: 0 } }>
+			<Icon icon={ production } />
+			<span>{ __( 'Production' ) }</span>
+		</HStack>
+	);
+};
+
+const CurrentEnvironment = ( { site }: { site: Site } ) => {
+	if ( site.is_wpcom_staging_site ) {
+		return <Environment env="staging" />;
+	}
+
+	return <Environment env="production" />;
+};
+
+const EnvironmentSwitcherDropdown = ( {
+	currentSite,
+	otherEnvironment,
+	otherEnvironmentSite,
+	onClose,
+}: {
+	currentSite: Site;
+	otherEnvironment: Env;
+	otherEnvironmentSite?: Site;
+	onClose: () => void;
+} ) => {
+	const productionSite = otherEnvironment === 'staging' ? currentSite : otherEnvironmentSite;
+	const stagingSite = otherEnvironment === 'staging' ? otherEnvironmentSite : currentSite;
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const mutation = useMutation( stagingSiteCreateMutation( productionSite?.ID ?? 0 ) );
+	const handleCreate = () => {
+		mutation.mutate( undefined, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Staging site created.' ), { type: 'snackbar' } );
```

Comment:
Since the staging site won't be immediately available, should we use the copy for the current experience?

<img width="571" height="72" alt="SCR-20250711-ntas" src="https://github.com/user-attachments/assets/717a8086-1d3a-44e2-bc34-b338a0b3b3de" />

cc: @matt-west @Marinatsu

Reply:
See the discussion on DOTDASH-107.